testkit/src/main/scala/org/bitcoins/testkit/core/gen/PSBTGenerators.scala

```diff
@@ -1,24 +1,199 @@
 package org.bitcoins.testkit.core.gen
 
 import org.bitcoins.core.config.BitcoinNetwork
+import org.bitcoins.core.number.UInt32
 import org.bitcoins.core.protocol.script.ScriptPubKey
 import org.bitcoins.core.protocol.transaction.TransactionOutput
-import org.bitcoins.core.psbt.{
-  GlobalPSBTMap,
-  GlobalPSBTRecord,
-  InputPSBTMap,
-  OutputPSBTMap,
-  PSBT
-}
+import org.bitcoins.core.psbt.GlobalPSBTRecord.Version
+import org.bitcoins.core.psbt.InputPSBTRecord.PartialSignature
+import org.bitcoins.core.psbt.PSBTInputKeyId.PartialSignatureKeyId
+import org.bitcoins.core.psbt._
 import org.bitcoins.core.wallet.builder.BitcoinTxBuilder
 import org.bitcoins.core.wallet.fee.FeeUnit
-import org.bitcoins.core.wallet.utxo.BitcoinUTXOSpendingInfoFull
+import org.bitcoins.core.wallet.signer.BitcoinSignerSingle.signSingle
+import org.bitcoins.core.wallet.utxo.{BitcoinUTXOSpendingInfoFull, UTXOSpendingInfoFull}
 import org.scalacheck.Gen
+import scodec.bits.ByteVector
 
+import scala.annotation.tailrec
 import scala.concurrent.{ExecutionContext, Future}
+import scala.jdk.CollectionConverters._
 
 object PSBTGenerators {
 
+  private def unknownGlobal: Gen[GlobalPSBTRecord.Unknown] = {
+    for {
+      key <- StringGenerators.hexString
+      value <- StringGenerators.hexString
+    } yield {
+      val keyBytes = ByteVector.fromHex(key).get
+      // Do a bunch of loops to guarantee that it is not a taken KeyId
+      @tailrec
+      def loop(bytes: ByteVector): ByteVector = {
+        if (bytes.size > 1 && PSBTGlobalKeyId.fromByte(bytes.head) != PSBTGlobalKeyId.UnknownKeyId) {
+          loop(bytes.tail)
+        } else {
+          bytes
+        }
+      }
+      @tailrec
+      def loop2(bytes: ByteVector): ByteVector = {
+        val newBytes = loop(bytes)
+        if (newBytes == ByteVector.empty) {
+          loop2(ByteVector(newBytes.hashCode.toByte))
+        } else {
+          newBytes
+        }
+      }
+      val usableKeyBytes = loop2(keyBytes)
+      GlobalPSBTRecord.Unknown(usableKeyBytes, ByteVector.fromHex(value).get)
+    }
+  }
+
+  private def unknownGlobals: Gen[Vector[GlobalPSBTRecord.Unknown]] = {
+    Gen.choose(0, 4).flatMap(num => unknownGlobals(num))
+  }
+
+  private def unknownGlobals(
+      num: Int): Gen[Vector[GlobalPSBTRecord.Unknown]] = {
+    Gen
+      .sequence(0.until(num).map(_ => unknownGlobal))
+      .map(_.asScala.toVector)
+  }
+
+  private def unknownInput: Gen[InputPSBTRecord.Unknown] = {
```

Comment:
You should have one private def that generates a random record like this whose key matches nothing in all three realms (or in just the one if you pass in the factory), and then interpret those bytes differently in three one-liner defs rather than repeating